### PR TITLE
Added optional cache-control to http headers

### PR DESCRIPTION
**Pull request by aish-where-ya**
## Feature or Problem
The user has an option to pass cache-control into the http header as a linked value.

## Related Issues
#238 

## Release Information
Next

## Consumer Impact
No drastic effect. Additional and optional parameter in linkdef value.

## Testing
Manual testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)
NA

### Acceptance or Integration
NA

### Manual Verification
Manually verified

### CODE REVIEW

File: httpserver-rs/src/lib.rs [hunk 1]
```diff
@@ -340,6 +351,7 @@ fn convert_response_headers(
                 continue;
             }
         };
+        map.remove(&name);
```
Comment (connorsmith256) [inline httpserver-rs/src/lib.rs:354]: Why was this needed?

File: httpserver-rs/src/settings.rs [hunk 2]
```diff
@@ -186,6 +191,13 @@ impl ServiceSettings {
                 }
             }
         }
+        if let Some(cache_control) = self.cache_control.as_ref() {
+            for val in cache_control.iter() {
+                if http::HeaderValue::from_str(val).is_err() {
+                    errors.push("Invalid Cache Control header".to_string());
```
Comment (connorsmith256) [inline httpserver-rs/src/settings.rs:197]: Should we include which one failed/the specific error message?

File: httpserver-rs/src/settings.rs [hunk 3]
```diff
@@ -33,6 +33,7 @@ const CORS_ALLOWED_HEADERS: &[&str] = &[
     "accept-language",
     "content-type",
     "content-language",
+    "cache-control",
```
Comment (connorsmith256) [inline httpserver-rs/src/settings.rs:36]: Are these case-sensitive? I noticed we used `Cache-Control` on the response

File: httpserver-rs/src/settings.rs [hunk 4]
```diff
@@ -252,6 +264,16 @@ pub fn load_settings(values: &HashMap<String, String>) -> Result<ServiceSettings
         ));
     }
 
+    // accept cache-control header values
+    if let Some(cache_control) = values.get("cache_control") {
```
Comment (connorsmith256) [inline httpserver-rs/src/settings.rs:268]: Let's update the settings documentation as well: https://github.com/wasmCloud/capability-providers/blob/main/httpserver-rs/settings.md

File: httpserver-rs/src/lib.rs [hunk 5]
```diff
@@ -230,7 +229,19 @@ impl HttpServerCore {
                                 http::StatusCode::INTERNAL_SERVER_ERROR
                             }
                         };
-                        *http_response.status_mut() = status;
+                        let http_builder = http::Response::builder()
+                        .status(status);
+                        let http_builder = if let Some(cache_control_header) = arc_inner.settings.cache_control.as_ref(){
+                            let mut builder = http_builder;
+                            for val in cache_control_header{
+                                builder = builder.header("Cache-Control",val)
```
Comment (connorsmith256) [inline httpserver-rs/src/lib.rs:237]: Just want to double-check that this appends vs overwrites if there are multiple

**Review by connorsmith256 — CHANGES_REQUESTED** [review]
Just a couple requests for updating docs/better error messages, and then a few questions for my own understanding.

Did we need to split the value and iterate because the http header struct didn't like things like `max-age=604800,must-revalidate`?

Also, how does the `=` interact with wash/the washboard? I.e. `wash ctl link put MXXX VXXX wasmcloud:httpserver cache_control=max-age=604800`

File: httpserver-rs/src/lib.rs (hunk 1 above)
Comment (aish-where-ya) [reply httpserver-rs/src/lib.rs:354]: So that if the value is set by the user in the response, that value takes precedence over the one in the linkef. 
@thomastaylor312 can you take a look as well.

File: httpserver-rs/src/settings.rs (hunk 3 above)
Comment (aish-where-ya) [reply httpserver-rs/src/settings.rs:36]: Headers are case insensitive

File: httpserver-rs/src/lib.rs (hunk 5 above)
Comment (connorsmith256) [reply httpserver-rs/src/lib.rs:237]: How does that translate to the response over HTTP? Can a header appear multiple times, or do these need to be concatenated?

File: httpserver-rs/settings.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ An empty tls section, or no tls section, disables tls. To enable TLS, both `cert
 - `allowed_headers` - a list of allowed headers, case-insensitive. See [`Access-Control-Allow-Headers`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers) Default:
```
Comment (connorsmith256) [inline httpserver-rs/settings.md:27]: GH isn't letting me comment on the file, but I think we need a top-level entry for `cache_control` now, right? These aren't intended to be set as part of the `cors` link def value, or are they?

File: httpserver-rs/src/lib.rs (hunk 5 above)
Comment (aish-where-ya) [reply httpserver-rs/src/lib.rs:237]: From what I know, these need to be comma separated instead of individual values. @thomastaylor312 @lachieh may know better.

Comment (lachieh) [reply httpserver-rs/src/lib.rs:237]: yah, comma separated joined in single header:

`Cache-Control: max-age=0, private, must-revalidate`

**Review by connorsmith256 — APPROVED** [review]
